features/consent/src/main/java/in/projecteka/jataayu/consent/viewmodel/ConsentHostFragmentViewModel.kt

```diff
@@ -0,0 +1,33 @@
+package `in`.projecteka.jataayu.consent.viewmodel
+
+import `in`.projecteka.jataayu.util.livedata.SingleLiveEvent
+import androidx.databinding.ObservableBoolean
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
+
+class ConsentHostFragmentViewModel() : ViewModel(), SwipeRefreshLayout.OnRefreshListener {
+
+    //refreshDataEvent
+    val refresh = MutableLiveData<Boolean>()
+    val isRefreshing = ObservableBoolean(false)
+
+    internal enum class Action {
+        ADD_FRAGMENTS
+    }
+
+    internal val viewPagerState = SingleLiveEvent<Action>()
+
+    fun setUp(){
+        viewPagerState.value = Action.ADD_FRAGMENTS
+    }
+
+    override fun onRefresh() {
+        isRefreshing.set(true)
+        refresh.value = true
+    }
+
+    fun setIsRefreshing(isRefresh: Boolean){
```

Comment:
since the property refresh is anyway public, getter and setter not required. Observables are immutable, but their value is not.